src/compiler/c.rs

```diff
@@ -151,8 +156,9 @@ impl Language {
 
     pub fn as_str(self) -> &'static str {
         match self {
-            Language::C => "c",
-            Language::Cxx => "c++",
+            Language::C | Language::CHeader => "c",
+            Language::Cxx | Language::CxxHeader => "c++",
```

Comment:
FYI this led to issue #1851 and to commit b881869902690240bd97872d4df2395c436f2806 to remediate.